### PR TITLE
Standardise all supervisor output logging

### PR DESCRIPTION
**Pull request by CameronDiver**
Also use the supervisor's own container logging monitoring code when
running livepush on the supervisor container.

Change-type: minor
Signed-off-by: Cameron Diver <cameron@balena.io>

### CODE REVIEW

File: src/lib/supervisor-console.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	// is what happens internally in winston)
+}) as unknown) as { [key in logLevel]: (message: string) => void };
+
+winston.addColors(colors);
+
+const messageFormatter = (printFn: (message: string) => void) => {
+	return (...parts: any[]) => {
+		let message = '';
+		let start = '';
+		for (const part of parts) {
```
Comment (Page-) [inline src/lib/supervisor-console.ts]: I think the gist of this function is
```ts
parts
	.map((p) => {
		if (p instanceof Error) {
			return p.stack
		}
		return p
	})
	.join(' ')
	.replace('\n','\n  ')
	.split('\n')
	.forEach((p) => {
		printFn(p)
	})
```
right?  Does `printFn` not handle multi-line messages very well?

File: src/lib/supervisor-console.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+			switch (typeof part) {
+				case 'string':
+					message += `${start}${part}`;
```
[the remark this answers is not in the document]
Comment (hedss) [reply src/lib/supervisor-console.ts]: Or if you want to be explicit, just:
```
case 'string':
default:
    ....
```

File: src/lib/supervisor-console.ts (hunk 1 above)
Comment (CameronDiver) [reply src/lib/supervisor-console.ts]: It breaks the formatting a little bit, but it's not the end of the world. This way just makes sure it's consistent. Will change to more functional, the idea was that I'd add more specific handlers for data types, but I didn't get there yet (For example a consistent way of showing docker output etc).

File: src/lib/supervisor-console.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
+import * as _ from 'lodash';
+import stripAnsi from 'strip-ansi';
+import * as winston from 'winston';
+
+const levels = {
```
[the remark this answers is not in the document]
Comment (CameronDiver) [reply src/lib/supervisor-console.ts]: Only that winston requires them in this format.

File: src/lib/supervisor-console.ts (hunk 1 above)
Comment (Page-) [reply src/lib/supervisor-console.ts]: You can still add more handlers in the
```ts
	.map((p) => {
		if (p instanceof Error) {
			return p.stack
		}
		return p
	})
```
btw

Comment (CameronDiver) [reply src/lib/supervisor-console.ts]: Yeah I know, I was mainly talking about the `switch`, which i will end up adding back in when/if we get more specific handlers.

**Pull request merged**